### ISSUE TITLE
jailbreak update 

### ISSUE THREAD

**Issue by misterjbam** [analoguejb/Analogue-Duo-JB#10]
Good morning

Would it be possible to update the jailbreak to have the same firmware version as the official one?

**Comment by ShaddamIV** [comment]
That would be nice.

**Comment by unixchaos** [comment]
Yes we are all big fans of your JB and if you could update it, that would be amazing.

**Comment by MireWolf82** [comment]
Awesome work.  Still, I'm holding off patiently for that 1.0 update.  To echo the others here, if it also includes the official firmware 1.3 (as of this comment) updates, that would rock.  Even if it isn't ready, and update on the update would be nice.

**Comment by theDarkFrenchy** [comment]
Don't want to sound harsh but.... no news since the first jailbreak....
I won't hold my breath and it's really sad, especially when you bought this system. 
Genuine PCE units are now really really old...  :-(

**Comment by MireWolf82** [comment]
Yes it feels like they threw it out there and the Fuo itself to be honest just to clear their promises.  Now they have, on to bigger more profitable things.  It's a shame, a 1.0 solid tested firmware with the bugs squashed would be awesome; both to save my Duo's lens/longetivety and save me having to buy the pricy Turbo Everdrive Pro.  (I already have the incompatible one)

**Comment by theDarkFrenchy** [comment]
I know the feeling... I guess Analogue is working on their N64 FGPA....
But it's pretty sad to see this !! The other solution possible is ... MisTer. 
But I still have my original games here....

**Comment by MireWolf82** [comment]
Yep definitely looking into the Mister Pi by Taki Udon as 500+ for a decked out Mister is a bit much...  Coming from a guy with 2 Analogue Pockets w/dock lol.

But I really was hoping the Duo would become the consolized de facto Open FPGA option seeing as though it uses Analugue OS.  Sega CD support for example what with the disc drive and all.

**Comment by theDarkFrenchy** [comment]
Me too...  I have the SNES and MD Analogue products.

**Comment by DeerSteak** [comment]
> Yep definitely looking into the Mister Pi by Taki Udon as 500+ for a decked out Mister is a bit much... Coming from a guy with 2 Analogue Pockets w/dock lol.
> 
> But I really was hoping the Duo would become the consolized de facto Open FPGA option seeing as though it uses Analugue OS. Sega CD support for example what with the disc drive and all.

It very much could have been the de facto standard, but Analogue chose not to give it openFPGA. My guess is that they regret that decision, since the Duo appears to not have sold well. It never went out of stock like the other products they've made.

**Comment by ShaddamIV** [comment]
It is a failure because it is still not update + jailbreak, it is either one or the other, but not both at the same time.
Analog must make sure their firmware is fully working before trying to sell a locked product.
Since most great games on the PC Engine are exclusive to Japan, there is no way for us to actually legally use that thing.

**Comment by DeerSteak** [comment]
Yeah, certainly seems like they've given up.

**Comment by cyxceven** [comment]
This still hasn't been updated to the latest fw for nearly a year. Open-source it, or make it a diff patch to the stock firmware.

**Comment by larsulrik** [comment]
@cyxceven I don't really understand the logistics of this project. Was the original firmware cracked by this team and then patched? How is it "legal" (not sure if that's the right word) to take a proprietary firmware, crack it, patch it, and put it up on Github? It should be on a community driven collaboration site IMHO to keep things moving as long as people are willing to work on it.

**Comment by MireWolf82** [comment]
I think at this point their hands are tied just making sure the 3D works well; the Duo sales don't justify any more effort on their part.  They got it to where it could run ROMs and that's it.  Im sure they're figuri g it's good enough, that if you want the update and ROMs, go get a Turbo Everdrive Pro; which is the vibe I get from them.  So yeah, gonna spend another 250 on this thing eventually for the pro, I've given up all hope on the jailbreak matching the firmware, so it's a take your pick situation.

**Comment by eewokee** [comment]
Not tied, just not willing to make it open source or release jailbreak firmware, because they are a commercial company. I switched to Misterfpga because of the lack of proper jailbreak firmware. It is commercially brilliant to sell ‘new’ products again and again based on similar fpga chips. Kevtris, their fpga coder knew better once, but it’s $ that is leading them all. Who can disagree. Just avoid buying ‘new’ products and buy a Misterfpga instead. Looks less fancy, but the system can do a lot more and is backed by a large community.

**Comment by larsulrik** [comment]
I figured the team would be passionate about retro gaming like we all are, regardless of the Analogue sales. The way I see it is that the JB team has a collection of Analogue jailbreaks that is a "line of products" in a way, that serves the new-retro console community. When the 3D comes out I think it would be a great time to update every jb with a new version. Easy for me to say of course as I leech their unpaid work. :)

I just hate to see these projects go cold and no one can really do anything about it.

**Comment by eewokee** [comment]
It is the same ‘team’. 😉

**Comment by larsulrik** [comment]
Wait, is the Analogue JB team also the Analogue team?? Am I misunderstanding this conversation? I'd feel stupid if that was true.

**Comment by misterjbam** [comment]
yes it is the analogue team that releases the jailbreak. no analogue console has been hacked, it is just the firmware recompiled with an additional option. that is why it is incomprehensible that there is no update.

**Comment by larsulrik** [comment]
Wow that's a first for me; the manufacturer releasing its own jailbreak. Does everyone know this?

**Comment by MireWolf82** [comment]
Honestly I'm good with the Analogue Pocket and dock.   Basically does the same thing.   With the Duo I at least get the tactile feel of putting the games in.  Which is really the point.  I mean I can burn CDs to play,  I just wasn't looking to put wear on the CD drive.  It IS for me a brand new console as I haven't used it much, so it is what it is.  I just wish I had the jail break AND the updated firmware.   I thunk I'll just update and get a Turbo Everdrive Pro somewhere down the line.  But hey in the off chance Kevtris reads this, an update to the jailbreak would be rad.  A non-1.0 updated jailbreak just feels off to me.  But hey it probably works just fine.

**Comment by Xorcist77** [comment]
Firmware 1.5 was just released... maybe we'll get lucky... *** crosses fingers *** and get a JB update....

**Comment by theDarkFrenchy** [comment]
The new firmware 1.5 really is interesting regarding the **controllers support** !!!   :-)
Analogue published a new firmware : 1.5 Date [2025-03-18]

MD5
6f99f692c2d9380ca0c5fae7bbb61e54

OS
Fixed Single-buffered Sync Mode being forgotten on power off

System
Added several missing HuCards to the Library
Fixed timing compatibility bug affecting some official and unofficial HuCards
Fixed reset bug on unofficial HuCards

**Controller
Added support for Switch Genesis/MD controllers
Added support for Switch NES/Famicom controllers
Added support for Buffalo Classic USB Gamepad
Added support for some later Wii Remote Plus controllers
Fixed button mapping for Switch Joy-Cons and Switch SNES/N64 controllers
Fixed various 8BitDo controllers when connected via USB: M30 2.4g, N30 2.4g, SN30 2.4g, Zero 2, and Micro
Fixed PS4 controller home button bug when connected via USB
Fixed occasional crash when attempting to pair controllers
Fixed reconnecting/pairing of Bluetooth LE controllers that had been paired to other devices**

**Comment by andreiborisov** [comment]
Literally waiting for the update to pull the plug on Analogue Duo…

**Comment by andreiborisov** [comment]
My educated guess is that they will wait for the last planned official update before releasing the updated JB version, simply because it would require much less work

**Comment by Rai** [comment]
Just adding my +1 to keep interest alive. I just got my Duo

**Comment by computerdoctor300** [comment]
> Just adding my +1 to keep interest alive. I just got my Duo

me too  lets get this updated :)

**Comment by ShaddamIV** [comment]
Like all Analog products, their weakness is software.
The Analog Duo was abandoned on day one.
Never again for me.

Get a Mister or else next time.

**Comment by andreiborisov** [comment]
> Like all Analog products, their weakness is software.

I’m not sure I can agree with that statement. Analogue Pocket has been great for me.

**Comment by RosySyn** [comment]
Considering how much has improved since the firmware this jailbreak is based on was released, it would be wonderful if we could see an updated jailbreak. If it's a monetary issue, I'd be happy to contribute - I'm sure this is true for many others as well. 

This jailbreak is the ONLY way to experience stereo audio when running CD games from a backup _without_ making a physical copy, and as such, I think this is vital. When running CD games through the Turbo EverDrive Pro, it's mono audio only, and there's no firmware/hardware solution possible on their end. I'd like to avoid making physical CD backups when possible, as optical media (and the hardware needed to access it) is notoriously unreliable...

As a side note, after doing a quick search regarding this topic, I'm positive the lack of an updated jailbreak is hurting Duo sales. Other popular FPGA consoles natively have the ability to do what this jailbreak does. Just wanted to point this out...

Much appreciation for this team's hard work! :)

**Comment by ztoday21** [comment]
I'm still here waiting.

**Comment by ShaddamIV** [comment]
This thing is useless without a jailbreak. The Everdrive Pro does not even works that well. I never bought another Analog product and I never will. When even Atari, C64 minis, Amiga Minis, Spectrum all offer to load ROMS from usb key, Analog is a company that has no understanding of the modern retro world we live in.

Never buy Analog products, they are scammers that never deliver. Analog Duo was abandoned on day one. And the new Analog 64 is crappy as hell with ugly interlaced image. There are better FPGA products that offer much more flexibility.

**Comment by andreiborisov** [comment]
The Duo situation is really unacceptable and we should demand the updated version of the jailbreak firmware.

**How about we all contact the Analogue Support (https://www.analogue.co/support/contact) and say, that we want the jailbreak firmware to be made up-to-date?** I highly doubt anyone from Analogue reads this thread, considering how busy they've been with 3D.

Here is an example:

```
Request for Updated Alternative Firmware for Analogue Duo

```

```
Hi,

I'm a Duo owner and while I'm enjoying the device, I'm sad that the alternative firmware (available on GitHub) hasn't been updated to the version 1.5, and therefore is missing ton of critical fixes and support for essential features (save states, etc.).

I wonder if it would be possible to release an updated version... Now, I know, that that firmware is unofficial and is not released by Analogue, but there should be something that can be done...

While you might make less direct financial incentive to support the Duo, considering it's a more of a niche device, I think it makes sense in a long run, since it affects customer perception and brand loyalty (there are a ton of people who didn't buy 3D yet, because they don't believe you will continue to support it after getting burned by Duo).

Best,
your name
```




@ShaddamIV Again, I completely agree with you that they abandoned Duo, and this sucks, however, calling Analogue scammers is very far-fetched.

If you want all-in-one solution there is MisterFPGA, Analogue products are just following a different philosophy (one core — one console). Ain't nothing wrong with that.

The interlacing issue in a small amount of N64 titles were just fixed btw.

**Comment by RosySyn** [comment]
As suggested, I have sent an email to Analogue support asking if there's any possibility that the jailbreak firmware will receive an update. Here's what I sent, just in case someone at Analogue stumbles upon this post (or if others would like to copy my message, which I do recommend!):

***

"Hello! I'm an Analogue Duo owner and, while I'm enjoying the device, it's sad to see that the alternative firmware (available on GitHub) hasn't been updated to version 1.5, and therefore is missing tons of critical fixes and support for essential features (CD save states, advanced controller remapping, etc.). It seems that all support has been dropped for this device, and I haven't seen any official statements stating otherwise. 

My question: Would be possible to release an updated version of this firmware? I know that the firmware is technically "unofficial" and was not released "directly" from Analogue (though it's clear it was created by an Analogue dev!), but there must be something that can be done? 

While it may seem like there's less direct financial incentive to support the Duo, considering it's a more of a niche device, I do think it makes sense to provide this support in the long run, since it affects customer perception and brand loyalty (there are a LOT of people who didn't buy the 3D yet, myself included, because they don't believe you will continue to support it, especially after getting burned by the Duo). Something to consider! I hope to see more support for Duo owners in the future."

***

It's been exactly one week, and I have yet to receive a response.

Again, I would gladly _pay_ for an update to the firmware, _an update that is likely trivial for them to release_, but without being able to get in touch with the creator of said firmware, or anyone at Analogue, **even as a paying customer**... I guess I have to give up and move on, or hope someone else in the community takes it upon themselves to release a patch (if that is even possible).

The retro hardware community thrives on the passion of kind enthusiasts who are willing to communicate and contribute - I really, really hope to see Analogue step up to the plate and make an effort to become _part_ of this community. The Duo deserves support!

EDIT: It's been over a month without a response. Unfortunately, Analogue doesn't seem to care about responding to paying customers.

**Comment by andreiborisov** [comment]
@rosysyntax nice! I don’t expect them to actually answer, it would be a potential legal liability to admit (even implicitly) they had released a jailbreak firmware.

What I expect is that they silently release the update if we write enough to their support.

**Comment by ShaddamIV** [comment]
There is nothing illegal about this, all the new mini consoles (even official Atari stuff) and mini computers can read roms from usb/sd-card. What they do cannot be illegal, what can be illegal is using a ROM that you don't own. None of that is their business. I did not buy the A64 bacause how stupid can you be to sell a product that cannot be used. 99% of potential buyers do not have any games. Most new games for the PC-Engine/N64 have no physical release. I think Analalog don't want to make money and they want to go bankrupt.

There is a new FPGA N64 coming and it will read games roms from sd-card/usb straight out of the box.

You can't fix stupid. Move on, give your money to intelligent people instead.  Analog is weird and stupid, they only make bad decisions.

This is not jailbreak, its just reading roms that you own or bought recently from a indie dev.

**Comment by andreiborisov** [comment]
> There is nothing illegal about this

There is nothing illegal about developing software emulators too, yet Nintendo managed to sue its developers into oblivion. Even if you're completely in the right, you'd inquire insane legal costs while trying to defend yourself in the court, the kind of cost that would make a small company like Analogue go bankrupt.

**Comment by ShaddamIV** [comment]
That is why it is important to make all software open source. Closed source code keeps us all in eternal darkness.

**Comment by SpiralBrad** [comment]
> That is why it is important to make all software open source. Closed source code keeps us all in eternal darkness.

Agreed. I think I'm done with Analogue.

**Comment by andreiborisov** [comment]
Well, the MisterFPGA is certainly out there and is an option, especially now when it's being more readily available than ever with stuff like SuperStation One, etc. As a UX engineer, I still find Analogue products way superior though. But I'm sure most people do not care about this stuff as much.

**Comment by RosySyn** [comment]
> @rosysyntax nice! I don’t expect them to actually answer, it would be a potential legal liability to admit (even implicitly) they had released a jailbreak firmware.
> 
> What I expect is that they silently release the update if we write enough to their support.

The creator of this firmware is literally [analoguejb](https://github.com/analoguejb). If legal trouble were to be had, it would have been done by now. I do not think the legality of providing this firmware is why Analogue has not provided an update - otherwise, they would not have released it at all in the first place (correct me if I'm wrong, Analogue staff!). 

I think the actual reason is far more simple - they don't think releasing an update would translate to sales, which is... misguided at best, and anti-consumer at worst. These consoles thrive on word of mouth, and supporting your customers is important, regardless of whether there's further profits to be made. I think that if they want to retain a positive image in our (expensive) hobbyist space, they really should make an attempt to _interact_ with the folks spending hundreds of dollars on their premium hardware.

As a side note, it's been a month since I sent my email to them. I never got a reply. Nothing. As a paying customer, that's extremely disappointing. _I'm not purchasing a product from a company that doesn't see the value in communicating with its consumers_. I will happily change my tune if they make things right, however - I'm not here to burn anyone at the stake, especially if this was simply the case of a missed ticket.

On the flipside, I'm very excited to be a future SuperStation One owner.

**Comment by ShaddamIV** [comment]
I did not buy the Analog 64 because Analog is a terrible company that abandon their products on day 1.
One of my friend did not buy an Analog Duo because I told him that it was abandoned on day 1.
They want to make money, they should start supporting their products.
Also many games still don't run properly on that thing.

Analog, you treat your customers like trash : we don't buy your products.
I think Analog only sell to people who never bought their products.
Once you know that their products are expensive trash you don't get scammed another time.

The Analog 64 is a broken product. Only paid youtubers shill for this company.

**Comment by Krathoon** [comment]
Yeah. I is kind of a shame that the jailbreak does not have the latest firmware. The latest firmare fixes problems with the CD drive not working right.
So, with the latest firmware, I just use the duo to play game discs for nostalgia.
